### DISCUSSION TITLE
AIP-66 Web Wallet Communication Method?

### DISCUSSION THREAD

**Discussion by wilwade** [aptos-labs/aptos-developer-discussions#183] [General]
Hello! I've been pulled in by looking at how Aptos handles passkeys/webauthn for accounts per AIP-66.

One piece of the puzzle I don't yet understand about AIP-66 is how a wallet website and a dApp website communicate.

Setup: Wallet Website `example-wallet.com`, dApp Website `dapp-example.com`

1. Alice visits `example-wallet.com` and creates an account, transfers some APT to her address to create the account on chain.
2. Alice then visits `dapp-example.com` and wants to use the wallet from `example-wallet.com`
3. How does `dapp-example.com` communicate with `example-wallet.com`?

I looked over the `aptos-wallet-adapter`, but it not clear there either how one might accomplish this setup.

The closest I got is how msafe does it via iframe, but that doesn't feel like the intended path.

Perhaps I am just missing something?

**Comment by hariria** [comment]
Typically it would be via iframe or popup. The wallet adapter is really for extension or mobile wallets

**Reply by wilwade** [reply]
Are there there any standards for that, or would it just be about adding your specific wallet discovery and iframe API into the `aptos-wallet-adapter`?

**Reply by hariria** [reply]
No plans to integrate with the wallet adapter at the moment. The dapp would integrate the web wallet directly
